### DISCUSSION TITLE
why is llama.cpp server so slow as compared to ollama?

### DISCUSSION THREAD

**Discussion by EduardTalianu** [ggml-org/llama.cpp#8413] [Q&A]
what options can I enable to make it faster in a no gpu environment?

**Comment by dspasyuk** [comment]
@EdwardDali  you can use --ngl N option to offload to GPU, where N is the number of layers you want to offload to GPU for 7B model it is around 30. Read this for more details https://github.com/ggerganov/llama.cpp/blob/master/examples/main/README.md or use llama-cli -h
